server/services/caldav/lib/calendar/calendar.cleanUp.js

```diff
@@ -0,0 +1,27 @@
+const logger = require('../../../../utils/logger');
+
+/**
+ * @description Delete all CalDAV calendars and events.
+ * @param {Object} userId - Gladys user to clean up.
+ * @returns {Promise} Resolving with client connected.
+ * @example
+ * cleanUp(user.id)
+ */
+async function cleanUp(userId) {
+  logger.info(`Start cleaning CalDAV data for user: ${userId}`);
+  const gladysCalendars = await this.gladys.calendar.get(userId, { serviceId: this.serviceId });
+  logger.info(`${gladysCalendars.length} calendars to clean`);
+  await Promise.all(
```

Comment:
Be careful with promise.all on big accounts, if you have lots of calendar, can be too much in the call stack... 

You can use bluebird Promise.map instead with a concurrency set